File: A's/263A.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int n = 5; // Must be an odd number

    int x = n / 2, y = n / 2; // Middle of the matrix
    int x1, y1;

    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n; j++) {
            int v; cin >> v;
            if (v == 1) {
                x1 = i;
                y1 = j;
            }
        }
    }

    int res = abs(x1 - x) + abs(y1 - y);
    cout << res;
    
    return 0;
}

File: A's/py/263A.py
def main() -> None:
    n: int = 5 # Must be an odd number to work as intended

    m: int = n // 2 # Middle of the matrix
    x: int = 0
    y: int = 0

    for i in range(n):
        row: list[str] = list(map(str, input().split()))
        if "1" in row:
            y = i
            x = row.index("1")

    res: int = abs(x - m) + abs(y - m)
    print(res)

if __name__ == "__main__":
    main()